operator/src/main/java/org/bf2/operator/managers/KafkaManager.java

```diff
@@ -428,12 +429,16 @@ private boolean isKafkaUpgradeStabilityCheckJobExists(ManagedKafka managedKafka)
     private void addUpgradeTimeStampAnnotation(ManagedKafka managedKafka, String annotation) {
         log.debugf("[%s/%s] Adding Kafka upgrade %s timestamp annotation",
                 managedKafka.getMetadata().getNamespace(), managedKafka.getMetadata().getName(), annotation);
+
+        String timestampValue = ZonedDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+        managedKafka.getMetadata().getAnnotations().put(annotation, timestampValue);
+
         managedKafkaClient
                 .inNamespace(managedKafka.getMetadata().getNamespace())
                 .withName(managedKafka.getMetadata().getName())
                 .edit(mk -> new ManagedKafkaBuilder(mk)
```

Comment:
based upon the modification the managedKafka above, this could just be .edit(mk -> managedKafka)

Reply:
@shawkins , I was actually questioning whether updating the MK in this class is the right thing to do. I believe it frequently results in HTTP/409 when the operator SDK goes to update the resource after the reconcile. I wouldn't want to do anything about that in this PR, but maybe it deserves some attention at some point.

Reply:
Yes, if this is happening in-line with the main reconciliation loop, then it's definitely wrong to issue this additional update.  Instead ManagedKafkaController should return UpdateControl.updateResourceAndStatus.

Reply:
Opened MGDSTRM-10094 to track this.